test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperInputExtensionsTest.cs

```diff
@@ -0,0 +1,324 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Mvc.TestCommon;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Core
+{
+    /// <summary>
+    /// Test the <see cref="HtmlHelperInputExtensions" /> class.
+    /// </summary>
+    public class HtmlHelperInputExtensionsTest
+    {
+        [Fact]
+        public void CheckboxHelpers_UsesSpecifiedExpression()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var checkboxResult = helper.CheckBox("Property1");
+
+            // Assert
+            Assert.Equal(
+                "<input id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[checkbox]]\" value=\"HtmlEncode[[true]]\" />" +
+                "<input name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[hidden]]\" value=\"HtmlEncode[[false]]\" />",
+                HtmlContentUtilities.HtmlContentToString(checkboxResult));
```

Comment:
@Eilon I agree in general but not for these tests. As-is, they aren't unit tests and are instead testing `DefaultHtmlGenerator`, `HtmlHelper`, `HtmlHelper<T>` as well as `HtmlHelperInputExtensions`. So I'd like to see lots of checks of the rendered HTML in unit tests of `HtmlHelper` and `HtmlHelper<T>`.

Could take that to an extreme and mock `IHtmlGenerator` when testing `HtmlHelper`. But we don't have many tests like that today and `IHtmlGenerator` is something of an implementation detail for HTML and tag helpers.